### PR TITLE
feat: add drum machine to new cert

### LINKED ISSUE #656: Add drum machine lab to main repo
Once the prototype has been approved, then we will add this lab to the main repo with tests.

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes https://github.com/freeCodeCamp/CurriculumExpansion/issues/656

### CODE REVIEW

File: curriculum/challenges/_meta/lab-drum-machine/meta.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Build a Drum Machine",
+  "usesMultifileEditor": true,
+  "dashedName": "lab-drum-machine",
+  "superBlock": "full-stack-developer",
+  "challengeOrder": [{ "id": "6762ec275cef87635acc4fe3", "title": "Build a Drum Machine" }],
+  "helpCategory": "JavaScript",
+  "isUpcomingChange": false,
+  "blockLayout": "link"
```
Comment (Dario-DC) [inline curriculum/challenges/_meta/lab-drum-machine/meta.json:9]:
```suggestion
  "blockLayout": "link",
  "blockType": "lab"
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by moT01** [comment]
Do we want to use the same challenge ID here as the existing drum machine project?

**Review by Dario-DC — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]

**Review by moT01 — COMMENTED** [review]
Pretty fun @ilenia-magoni 🎉 

<details><summary>Here's the code I used to pass</summary>

```html
<!doctype html>
<html lang="en">

<head>
  <meta charset="utf-8" />
  <title>Drum Machine</title>
  <link rel="stylesheet" href="styles.css" >
</head>

<body>
  <div id="drum-machine">
    <p id="display"></p>
    <div id="pad-bank">
      <div class="drum-pad">Q<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Heater-1.mp3" id="Q"></div>
      <div class="drum-pad">W<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Heater-2.mp3" id="W"></div>
      <div class="drum-pad">E<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Heater-3.mp3" id="E"></div>
      <div class="drum-pad">A<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Heater-4_1.mp3" id="A"></div>
      <div class="drum-pad">S<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Heater-6.mp3" id="S"></div>
      <div class="drum-pad">D<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Dsc_Oh.mp3" id="D"></div>
      <div class="drum-pad">Z<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Kick_n_Hat.mp3" id="Z"></div>
      <div class="drum-pad">X<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/RP4_KICK_1.mp3" id="X"></div>
      <div class="drum-pad">C<audio class="clip" src="https://cdn.freecodecamp.org/curriculum/drum/Cev_H2.mp3" id="C"></div>
    </div>
  </div>
  <script src="script.js"></script>
</body>

</html>
```

```css
* {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  background-color: #3b3b4f;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
}

#display {
  width: 400px;
  height: 60px;
  color:  #d0d0d5;
  border: 4px solid #0a0a23;
  border-radius: 15px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 40px;
  font-family: sans-serif;
  margin: 0 0 20px 0;
}

#pad-bank {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  width: 400px;
  height: 400px;
  padding: 30px;
  border: 4px solid #0a0a23;
  border-radius: 15px;
  background-color: #3b3b4f;
}

.drum-pad {
  cursor: pointer;
  justify-self: center;
  align-self: center;
  padding: 30px;
  font-size: 40px;
  width: 80px;
  height: 80px;
  font-family: sans-serif;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #d0d0d5;
  background-color:  #2a2a40;
  border: 2px solid #0a0a23;
  border-radius: 5px;
}
```

```js
const drumPadEls = document.querySelectorAll('.drum-pad');
const displayEl = document.getElementById('display');
const validKeys = ['Q', 'W', 'E', 'A', 'S', 'D', 'Z', 'X', 'C'];

drumPadEls.forEach(el => {
  el.addEventListener('click', e => {
    e.target.firstElementChild.currentTime = 0;
    e.target.firstElementChild.play();
    displayEl.innerText = `${e.target.firstElementChild.id}!`
  });
});

document.addEventListener('keydown', function (event) {
  const key = event.key.toUpperCase();
  if (validKeys.includes(key)) {
    const audioEl = document.getElementById(key)
    audioEl.currentTime = 0;
    audioEl.play();
    displayEl.innerText = `${key}!`
  }
});
```

</details>

Didn't really run into any issues. Left one suggestion - A couple other things I wrote down...

We could maybe include the URL to the sound files so people don't have to open and copy it? Feels like that would be a lot of text - maybe there's a way to just include the base URL once. Maybe something like, "audio samples you can use for your drum machine can be found at `https://cdn.freecodecamp.org/curriculum/drum/<filename>.mp3` - then a small table with the titles you have and the filename to go with it? I noticed the titles don't match the filename - so maybe it's just best how you have it. I think it's fine as it is, but feel free to make adjustments if you want.

The other thing I wrote down is that I noticed a lot of "you should have another `div` with an id of `pad-bank`" in the user stories and test text. Not sure if this is common in the other labs and challenges, but I would either use `ID` (no backticks) if we want it as an abbreviation for "identification", or  `` `id` `` (backticks) if we want it as the `id` attribute. Either way works, but the lowercase id doesn't seem right to me without the code highlight. Pretty minor, I would be fine without changing those as well. Your call. I think I would vote to use ID so there isn't as many code blocks.

**Comment by moT01** [comment]
oh geez, looks like I'm late - lol

**Pull request merged**